addons/product/models/product_product.py

```diff
@@ -683,6 +683,8 @@ def _get_contextual_price(self):
             return 0.0
 
         quantity = self.env.context.get('quantity', 1.0)
+        currency = pricelist.currency_id or self.env.context.get('currency_id')
```

Comment:
Where do you specify this context ?
Also, it's confusing to put a `res.currency` record as `currency_id` in context, should be `currency` (or you pass the id and you convert it here)

Reply:
Ha, yes, there is a note for this if I remember, I know YTI want to get rid if this part, where we pass info through the context. (Here : https://github.com/odoo/odoo/blob/ef9e2d9f8bb1682ca9614769f17d2afc168254e5/addons/product/models/product_product.py#L707)
This context is added in `website_sale`